### PR TITLE
4 rasberry pi wifi upload script

### PR DESCRIPTION
**Pull request by GHAFHA**

### CODE REVIEW

File: mongo_client/mongo_client.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    def __init__(self, connection_string: str):
+        self.connection_string = connection_string
+        self.clientclient = MongoClient(
+            connection_string, server_api=ServerApi('1'))
+
+    def establish_connection(self):
+        try:
+            self.client.admin.command('ping')
+            print("Pinged your deployment. You successfully connected to MongoDB!")
```
Comment (Squidtoon99) [inline mongo_client/mongo_client.py:18]: We should implement proper logging but this should be fine for now

Comment (GHAFHA) [reply mongo_client/mongo_client.py:18]: yea we can use logging, i was gonna have vasu implement it

File: mongo_client/mongo_client.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    def close_connection(self):
+        self.client.close()
+        print("MongoDB connection closed.")
+
+    def insert_document(self, db_name: str, collection_name: str, data: dict):
+        try:
+            db = self.client[db_name]
+            collection = db[collection_name]
+
+            inserted_ids = collection.insert_many(data).inserted_ids
```
Comment (Squidtoon99) [inline mongo_client/mongo_client.py:31]: Why is this called insert document when we call insert many in the backend? This function screws with type checking and should be taking in named arguments too.

File: tests/conftest.py [hunk 3]
```diff
@@ -1,7 +1,10 @@
-from box_client.box_client import Box_Client
+from box_client.box_client import Client as Box_Client
+from mongo_client.mongo_client import Client as Mongo_Client
 import pytest
 import os
 
+uri = f"mongodb+srv://noel:${os.getenv('MONGO_PASSWORD')}@cluster0.gw7z3sn.mongodb.net/?retryWrites=true&w=majority"
```
Comment (Squidtoon99) [inline tests/conftest.py:6]: Noooooo don't do this. ENV variable *can* be the entire variable

File: tests/conftest.py [hunk 4]
```diff
@@ -11,3 +14,10 @@ def box_client():
         file_path="/Users/noeljohnson/Documents/file.txt",
```
Comment (Squidtoon99) [inline tests/conftest.py:14]: I get that this is testing but this should be a relative not absolute path

**Review by Squidtoon99 — CHANGES_REQUESTED** [review]
I think that we don't need an abstraction layer on top of the python MongoClient, we should instantiate a client, verify the connection is successful, and then return the same client.

File: tests/conftest.py (hunk 3 above)
Comment (GHAFHA) [reply tests/conftest.py:6]: i was lazy

File: mongo_client/mongo_client.py (hunk 2 above)
Comment (GHAFHA) [reply mongo_client/mongo_client.py:31]: i think we should have an abstraction layer on top of it no? keep it object oriented, mongo_client should be called in handler, and its methods used, we can also verify the methods work as intended with way using pytest

File: tests/conftest.py (hunk 4 above)
Comment (GHAFHA) [reply tests/conftest.py:14]: again lazy

File: mongo_client/mongo_client.py (hunk 2 above)
Comment (GHAFHA) [reply mongo_client/mongo_client.py:31]: i dont think renaming it is hard, i asked vasu to make the method more suitable as well as test

Comment (GHAFHA) [reply mongo_client/mongo_client.py:31]: and also if we're going to use logging id wanna keep the abstraction layer, keeps handler cleaner

**Comment by GHAFHA** [comment]
tasks to be done before closing pull request:
- [ ] Refactor box client code
- [ ] remove network code
- [ ] implement a proper handler
- [ ] figure out ssid, device_id, off pi, as well as the name of the network in the shop for the wifi-cliend, and the password for the network
- [ ] piece together handler with the correct information

**Pull request merged**